distribution/test/itests/test-itests-alliance/src/test/java/org/codice/alliance/test/itests/NsiliEndpointTest.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    @Test
+    public void testStandingQueryMgr() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        sampleNsiliClient.testStandingQueryMgr();
+
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testCatalogMgrGetHitCount() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        assertThat(sampleNsiliClient.getHitCount(), is(1));
+
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testSubmitQuery() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        // CatalogMgr
+        DAG[] results = sampleNsiliClient.submit_query();
+        assertThat(results.length, is(1));
+        DAG result = results[0];
+        assertThat(sampleNsiliClient.getProductIdFromDag(result), is(ingestedProductId));
+
+        // ProductMgr
+        DAG parametersDag = sampleNsiliClient.get_parameters(result);
+        assertThat(getAttributeFromDag(parametersDag, NsiliConstants.IDENTIFIER), is(ingestedProductId));
+        assertThat(getAttributeFromDag(parametersDag, NsiliConstants.STATUS), is("NEW"));
+        assertThat(sampleNsiliClient.get_related_file_types(result), is(new String[]{NsiliConstants.THUMBNAIL_TYPE}));
+        final String expectedThumbnailFilename = ingestedProductId + "-THUMBNAIL.jpg";
+        assertThat(sampleNsiliClient.get_related_files(result), is(new String[]{expectedThumbnailFilename}));
+        assertStubServerPutRequest(expectedThumbnailFilename, "image/jpeg");
+
+        // OrderMgr
+        String orderResultFilename = sampleNsiliClient.testOrder(result);
+        // TODO test processing packageElements of the order response https://codice.atlassian.net/browse/CAL-192
+        assertStubServerPutRequest(orderResultFilename, "application/x-tar");
+
+        deleteMetacard(ingestedProductId);
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testCatalogMgrViaCallback() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        sampleNsiliClient.testCallbackCatalogMgr();
+
+        sampleNsiliClient.cleanup();
+    }
+
+    private void startHttpListener() {
+        server = new StubServer(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort())).run();
+        whenHttp(server).match(method(Method.PUT), Condition.startsWithUri(NSILI_FILE_URI_PATH))
+                .then(Action.success());
+    }
+
+    private void assertStubServerPutRequest(String expectedFilename, String expectedType) {
+        assertThat(server.getCalls().size(), is(1));
```

Comment:
We should be able to use `verifyHttp()` instead of getting the calls and doing asserts on those. See https://github.com/mkotsur/restito#quick-example.